mono/metadata/gc.c

```diff
@@ -681,6 +682,31 @@ ves_icall_System_GCHandle_GetTarget (guint32 handle)
 	return mono_gchandle_get_target (handle);
 }
 
+static gboolean
+is_object_pinnable (MonoObject *obj)
```

Comment:
I've made the changes you suggested locally. Since this branch is so far behind master now, I'm going to create a new branch of master, make the changes there, and open a new PR.